packages/catalog-process/src/services/catalogService.ts

```diff
@@ -218,7 +218,17 @@ export function catalogServiceBuilder(
       const eService = await readModelService.getEServiceById(eServiceId);
 
       await repository.createEvent(
-        updateEserviceLogic({ eService, eServiceId, authData, eServiceSeed })
+        updateEserviceLogic({
+          eService,
+          eServiceId,
+          authData,
+          eServiceSeed,
+          eServiceWithSameName:
+            await readModelService.getEServiceByNameAndProducerId({
+              name: eServiceSeed.name,
+              producerId: authData.organizationId,
+            }),
```

Comment:
in theory this check should be inside the `updateEserviceLogic` function, because it's part of the business logic

Reply:
I agree, but there are few considerations. The `_Logic` functions don't contain straightforward access to the `readmodel`, so I initially passed `readModelService.getEServiceByNameAndProducerId` as a parameter, so it could be accessed from inside the `_Logic` function. Then, we refactored to the current change to make it simpler. If you prefer the previous change, I can revert back to that.

Reply:
it could be an improvement, considering that the logic could return an error before using the `eServiceWithSameName`, and in that case we could avoid the query.

I see that other logics use a similar approach, out of curiosity: any reason why the *Logics do not take the read model as dependency, doing the queries themself?

Reply:
I know, I completely agree with your points.
The reason why `_Logic` functions don't have straightforward access to the `readmodel` is that they were designed to be tested with unit tests, so that it's easier to mock a single query function, instead of mocking the entire `readModelService`. With this setup, two approaches can be followed: either passing the specific query to the `_Logic` function, or retrieving all the data in advance. By the way, this is an assumption, because I didn't design the functions at the beginning (cc @MalpenZibo @Viktor-K).

Anyway, now that the unit tests are going to be replaced by the integration tests, we could see if there's no longer the need to split the functions (example `cloneDescriptor` and `cloneDescriptorLogic`), because the former only retrieves the data for the latter, and in some cases it's more useful to retrieve additional data only after some business logic checks, as you are highlighting. By unifying the functions, I think we will improve readability as well.

Reply:
I agree, we could probably remove a couple of abstractions.
Anyway this is a low priority task, let's first complete activities required for the deployment and then in case refactor.
As of now, your suggestion of passing the function works for me

Reply:
Updated, thanks!